packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -78,15 +84,17 @@ class SocketReference extends TypedEventEmitter<ISocketEvents> {
      * Once the ref count hits 0, the socket is disconnected and removed
      * @param key - socket reference key
      * @param isFatalError - true if the socket reference should be removed immediately due to a fatal error
+     * @param avoidSocketClose - Caller suggesting not to close socket.
      */
-    public removeSocketIoReference(isFatalError: boolean) {
+    public removeSocketIoReference(isFatalError: boolean, avoidSocketClose?: boolean) {
         assert(this.references > 0, 0x09f /* "No more socketIO refs to remove!" */);
         this.references--;
 
         // see comment in disconnected() getter
         this.isPendingInitialConnection = false;
 
-        if (isFatalError || this.disconnected) {
+        // In case the socket is disconnected we still want to close it.
+        if ((isFatalError && !avoidSocketClose) || this.disconnected) {
```

Comment:
I think condition here is wrong. It should be (isFatalError | !avoidSocketClose)
And maybe we should not have to flags telling us the same thing - caller should do OR operation and pass just one flag into here.

Reason for that - isFatalError means socket is not usable. Today the only reason for that - we use multi-plexing and service does not support multi-plexing. It's a game over situation (and today it most likely means we connected to the wrong service).